fix(trimer): Fix IL2026 JSON warnings in src/Uno.UI

jonpryor opened the pull request:
Context: https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/prepare-libraries-for-trimming
Context: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation#specify-options

When building `src/Uno.UI` with trimmer warnings enabled (and turned up to 11):

	dotnet build  src/Uno.UI/Uno.UI.Skia.csproj -bl \
	  -p:TrimmerSingleWarn=false -p:_ExtraTrimmerArgs=--verbose -p:IsTrimmable=true -p:IsAotCompatible=true

Many of the warnings are around System.Text.Json usage:

	…/src/Uno.UI/UI/Xaml/FontManifestHelpers.cs(34,6): error IL2026:
	  Using member 'System.Text.Json.JsonSerializer.Deserialize<TValue>(Stream, JsonSerializerOptions)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code.
	  JSON serialization and deserialization might require types that cannot be statically analyzed.
	  Use the overload that takes a JsonTypeInfo or JsonSerializerContext, or make sure all of the required types are preserved.

Update `JsonSerializer.Deserialize<TValue>()` usage to remove the IL2026 warnings.

Specifically, update `CoreWebView2WebMessageReceivedEventArgs.cs` to use `JsonSerializer.Deserialize<string>(string, JsonTypeInfo<string>)`, providing a `JsonTypeInfo<string>` derived from a
`JsonSerializerOptions` instance that only supports strings.

Update `FontManifest*.cs` to use
[`JsonSourceGenerationOptionsAttribute`][0] so that System.Text.Json source generation can generate an appropriate `JsonSerializerOptions`. This allows us to use
`JsonSerializer.Deserialize<FontManifest>(Stream, JsonSerializerContext)`.

[0]: https://learn.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonsourcegenerationoptionsattribute?view=net-9.0

**GitHub Issue:** closes #

<!-- Link to relevant GitHub issue if applicable. All PRs should be associated with an issue (GitHub issue or internal), unless the change is documentation related. -->

## PR Type:

<!--
Copy the labels that apply to this PR and paste them above:

- 🐞 Bugfix
- ✨ Feature
- 🎨 Code style update (formatting)
- 🔄 Refactoring (no functional changes, no api changes)
- 🏗️ Build or CI related changes
- 📚 Documentation content changes
- 🤖 Project automation
- 💬 Other... (Please describe)

-->


## What is the current behavior? 🤔

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->


## What is the new behavior? 🚀

<!-- Please describe the new behavior after your modifications. -->

## PR Checklist ✅

Please check if your PR fulfills the following requirements:

- [ ] 📝 Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.
- [ ] 🧪 Added [Runtime tests, UI tests, or a manual test sample](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] 📚 Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] 🖼️ Validated PR `Screenshots Compare Test Run` results.
- [ ] ❗ Contains **NO** breaking changes

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below.
     Please note that breaking changes are likely to be rejected -->

## Other information ℹ️

<!-- Please provide any additional information if necessary -->

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21722/docs/index.html

unodevops commented:
⚠️⚠️ The build [180507](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=180507) has failed on Uno.UI - CI.

jonpryor committed:
fix(trimer): Fix IL2026 JSON warnings in src/Uno.UI

Context: https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/prepare-libraries-for-trimming
Context: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation#specify-options

When building `src/Uno.UI` with trimmer warnings enabled (and turned up to 11):

	dotnet build  src/Uno.UI/Uno.UI.Skia.csproj -bl \
	  -p:TrimmerSingleWarn=false -p:_ExtraTrimmerArgs=--verbose -p:IsTrimmable=true -p:IsAotCompatible=true

Many of the warnings are around System.Text.Json usage:

	…/src/Uno.UI/UI/Xaml/FontManifestHelpers.cs(34,6): error IL2026:
	  Using member 'System.Text.Json.JsonSerializer.Deserialize<TValue>(Stream, JsonSerializerOptions)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code.
	  JSON serialization and deserialization might require types that cannot be statically analyzed.
	  Use the overload that takes a JsonTypeInfo or JsonSerializerContext, or make sure all of the required types are preserved.

Update `JsonSerializer.Deserialize<TValue>()` usage to remove the
IL2026 warnings.

Update `FontManifest*.cs` to use
[`JsonSourceGenerationOptionsAttribute`][0] so that System.Text.Json
source generation can generate an appropriate `JsonSerializerOptions`.
This allows us to use
`JsonSerializer.Deserialize<FontManifest>(Stream, JsonSerializerContext)`.

*Note*: two runtime errors came up while making this change, and my
Google-Fu didn't provide good answers.  For posterity, the errors and
their fixes:

Firstly, an intermediate form of this PR would throw from
`JsonSerializer.Deserialize<FontManifest>(…)`:

	Unhandled exception. System.InvalidOperationException: TypeInfoResolver 'FontManifestSerializerContext' did not provide property metadata for type 'FontManifest'.

The fix for this is that
[`JsonSourceGenerationOptionsAttribute.GenerationMode`][1] needs to
specify *both* `.Metadata | .Serialization`:

	[JsonSourceGenerationOptions(
	  GenerationMode = JsonSourceGenerationMode.Metadata | JsonSourceGenerationMode.Serialization)]

Secondly, an intermediate form of this PR would throw from
`JsonSerializer.Deserialize<FontManifest>(…)`:

	Unhandled exception. System.Text.Json.JsonException: The JSON value could not be converted to FontStyle. Path: $.fonts[0].font_style | LineNumber: …

The default enum deserialization behavior is to use the enum member
integral values, e.g. `2` for `FontStyle.Italic`, so it throws when
the string form is encountered:

	{ "font_style": "italic" }

The fix for this is to set
[`JsonSourceGenerationOptionsAttribute.UseStringEnumConverter`][2]=true.

[0]: https://learn.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonsourcegenerationoptionsattribute?view=net-9.0
[1]: https://learn.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonsourcegenerationoptionsattribute.generationmode?view=net-9.0
[2]: https://learn.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonsourcegenerationoptionsattribute.usestringenumconverter?view=net-9.0

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21722/docs/index.html

unodevops commented:
⚠️⚠️ The build [180745](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=180745) has failed on Uno.UI - CI.

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21722/docs/index.html

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21722/wasm-skia-net9/index.html

unodevops commented:
⚠️⚠️ The build [182378](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=182378) has failed on Uno.UI - CI.
